### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [openvenues/libpostal#23]
Would you like to add more error handling for return values from functions like the following?
- [malloc](http://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html "Memory allocation") ⇒ [KHASH_SORT_BY_VALUE](https://github.com/openvenues/libpostal/blob/ebaedb6bcf864f3cea26f1c7cf6239e99feae66c/src/collections.h#L57)
- [strdup](http://pubs.opengroup.org/onlinepubs/9699919799/functions/strdup.html "Duplicate a string.") ⇒ [feature_counts_update_or_add](https://github.com/openvenues/libpostal/blob/ebaedb6bcf864f3cea26f1c7cf6239e99feae66c/src/features.c#L29)

**Comment by thatdatabaseguy** [comment]
Good point. I'll be doing a full audit of error checking over the next few days. There are also some checks missing in the vector implementation. Thanks for bringing it up and if you have pull requests, don't hesitate to send them my way.

**Comment by elfring** [comment]
I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Unchecked return values") a bit more.

Are you interested to apply [aspect-oriented software development](http://aspectc.org/ "AspectC++")?
How do you think about to encapsulate error detection and corresponding exception handling as [a reusable aspect](http://en.wikipedia.org/wiki/Aspect_%28computer_programming%29 "Combination of a pointcut with advice") in your software?
